http-client/src/main/java/io/airlift/http/client/HttpUriBuilder.java

```diff
@@ -55,7 +57,7 @@ private HttpUriBuilder(URI previous)
         scheme = previous.getScheme();
         host = previous.getHost();
         port = previous.getPort();
-        path = percentDecode(previous.getRawPath());
+        path = isNull(previous.getRawPath()) ? null : percentDecode(previous.getRawPath());
```

Comment:
These methods are intended for use as lambdas. Using them directly makes the code harder to read since it's not the standard way to write null checks (which have been in Java forever).